### PR TITLE
Use list comprehensions to preserve order.

### LINKED ISSUE #148: encode_rows returns reverse of input
```
	
(dev1@127.0.0.1)2> riak_pb_ts_codec:encode_rows([[<<"family1">>, <<"series1">>, 1234567, <<"hot">>, 23.5], [<<"family2">>, <<"series99">>, 1234567, <<"windy">>, 19.8]]).
[{tsrow,[{tscell,<<"family2">>,undefined,undefined,
                 undefined,undefined,[],undefined,undefined,undefined},
         {tscell,<<"series99">>,undefined,undefined,undefined,
                 undefined,[],undefined,undefined,undefined},
         {tscell,undefined,1234567,undefined,undefined,undefined,[],
                 undefined,undefined,undefined},
         {tscell,<<"win
dy">>,undefined,undefined,undefined,undefined,
                 [],undefined,undefined,undefined},
         {tscell,undefined,undefined,undefined,undefined,undefined,
                 [],undefined,undefined,19.8}]},
 {tsrow,[{tscell,<<"family1">>,undefined,undefined,undefined,
                 undefined,[],undefined,undefined,undefined},
         {tscell,<<"series1">>,undefined,undefined,undefined,
                 undefined,[],undefined,undefined,undefined},
         {tscell,undefined,1234567,undefined,undefined,undefined,[],
                 undefined,undefined,undefined},
         {tscell,<<"hot">>,undefined,undefined,undefined,undefined,
                 [],undefined,undefined,undefined},
         {tscell,undefined,undefined,undefined,undefined,undefined,
                 [],undefined,undefined,23.5}]}]
```

See also RTS-421 in JIRA.

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #148

### CODE REVIEW

**Comment by lukebakken** [comment]
@hmmr I don't know if this reversal has any side-effects, but I figured we want data encoded in the same order as what comes in. Thanks for a review.

**Comment by hmmr** [comment]
To put it in context: There is a chain of `[NewElement | AlreadyAccumulated]` constructs (which reverse the order) instead of straight list comprehensions (which preserve order), with the net effect that the client miraculously sees the right order on a `get` or `query`, whereas the records are written backwards on the metal in leveldb. And there is a half-assed attempt to fix this in https://github.com/basho/riak_pb/pull/140 (which appears to be not sufficient).

In your PR, what's missing is the complementary reversal on the way back. Once you are fixing cases of appending-to-list-head in `encode_` functions, can you do so also in the `decode_` counterparts?

**Comment by lukebakken** [comment]
OK I'll get right to it, thanks for noticing the `decode_` bit.

**Comment by lukebakken** [comment]
@hmmr - all set. The `decode_` functions were preserving order by using `lists:reverse` but I simplified `decode_rows` to use a comprehension. `decode_cells` uses a reverse so I left that one alone due to all of the record pattern matching.

File: src/riak_pb_ts_codec.erl [hunk 1]
```diff
@@ -36,7 +36,6 @@
          encode_rows/1,
          encode_cells/1,
          decode_rows/1,
-         decode_row/1,
```
Comment (lukebakken) [inline src/riak_pb_ts_codec.erl]: Unused function

Comment (hmmr) [inline src/riak_pb_ts_codec.erl]: It is actually used, in riak-erlang-client (which shares riak_pb as a dependency with riak proper).

Comment (lukebakken) [inline src/riak_pb_ts_codec.erl]: Really? I recompiled my clone of Riak on the `end-to-end/timeseries` branch while using this branch of `riak_pb` and it worked fine. Grepping `riak-erlang-client` on the `end-to-end/timeseries` branch doesn't turn anything up either:

```
lbakken@brahms ~/Projects/basho/riak-erlang-client (end-to-end/timeseries=)
$ git pull && git grep '\<decode_row\>'
Current branch end-to-end/timeseries is up to date.
```

Comment (lukebakken) [inline src/riak_pb_ts_codec.erl]: `decode_rows` is used in `riak-erlang-client`:

```
lbakken@brahms ~/Projects/basho/riak-erlang-client (end-to-end/timeseries=)
$ git grep '\<decode_rows\>'
src/riakc_ts.erl:82:            [tuple_to_list(X) || X <- riak_pb_ts_codec:decode_rows(Response#tsgetresp.rows)]
src/riakc_ts_query_operator.erl:53:    Rows = riak_pb_ts_codec:decode_rows(Rows_),
```

**Pull request closed without merging**